src/components/Home/EventsTable/index.jsx

```diff
@@ -77,8 +77,9 @@ class EventsTable extends React.Component {
   }
 
   onDetailsClickHandler(townhall) {
-    populateEventModal(townhall)
-    $('.event-modal').modal('show');
+    if(this.props.selectTownhall) {
+      this.props.selectTownhall(townhall);
```

Comment:
I haven't tried running this locally yet, but I don't see where this is being cleared out. I would think when you close the modal this should go back to null

Reply:
It's not, wasn't sure how you would like it, so I'll have it clear on modal close.